app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalController.php

```diff
@@ -54,6 +58,11 @@ public function show(): JsonResponse
         $favicon = $settings->getSettingsPropertyModel('portal.favicon')
             ->getFirstMedia('portal_favicon');
 
+        if (! $settings->knowledge_management_portal_requires_authentication && ! Session::exists('guest_id') && ! Auth::guard('contact')->check() && ArticleWasHelpful::active()) {
```

Comment:
We should do this regardless of `$settings->knowledge_management_portal_requires_authentication` so let's remove that check from this statement.

I would also move the Feature Flag check to the beginning of this statement. In almost all scenarios you want to do FF checks first.

You also probably want to check the auth guard before the session

Reply:
Done